### PR TITLE
hvac swing mode

### PR DESCRIPTION
**Pull request by Noneangel**
I would like to use the [swing mode for hvac option of Home Assistant](https://developers.home-assistant.io/docs/core/entity/climate/#swing-modes). So I propose to add a `Climate.withSwingMode` methods.

### CODE REVIEW

**Comment by Koenkk** [comment]
Which device support this?

**Comment by Noneangel** [comment]
See https://github.com/Koenkk/zigbee-herdsman-converters/pull/4560#issuecomment-1234694112 for details. Right now, no device support it. If you accept this request, I will add it to https://github.com/Koenkk/zigbee-herdsman-converters/pull/4560

File: lib/exposes.js [hunk 1]
```diff
@@ -423,6 +423,14 @@ class Climate extends Base {
         return this;
     }
 
+    withSwingMode(modes, access=a.ALL) {
+        assert(!this.endpoint, 'Cannot add feature after adding endpoint');
+        const allowed = ['off', 'on', 'vertical', 'horizontal', 'both'];
```
Comment (Koenkk) [inline lib/exposes.js:428]: Looking at the HA docs, there does not seem to be a limitation on the allowed values. I propose that we also don't put any limitations (https://www.home-assistant.io/integrations/climate.mqtt/)

Comment (Noneangel) [reply lib/exposes.js]: I use `Climate.withFanMode` as template. It is not limited by [home-assistant](https://developers.home-assistant.io/docs/core/entity/climate/#fan-modes) but is limited [here](https://github.com/Koenkk/zigbee-herdsman-converters/blob/852b871de63bc557a850435f6c5fcdd6602f4ca6/lib/exposes.js#L420).

**Pull request merged**